numpy/ma/core.py

```diff
@@ -2756,13 +2756,19 @@ def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
                     _data._sharedmask = True
         else:
             # Case 2. : With a mask in input.
-            # Read the mask with the current mdtype
-            try:
-                mask = np.array(mask, copy=copy, dtype=mdtype)
-            # Or assume it's a sequence of bool/int
-            except TypeError:
-                mask = np.array([tuple([m] * len(mdtype)) for m in mask],
-                                dtype=mdtype)
+            # If mask is boolean, create an array of True or False
+            if mask is True:
+                mask = np.ones(_data.shape, dtype=mdtype)
+            elif mask is False:
+                mask = np.zeros(_data.shape, dtype=mdtype)
```

Comment:
Ok, I will a test for `mdtype`. For the unit tests it is covered by tests which use mask=True or False in the constructor, but maybe I can a test specific to the constructor.